src/coreclr/jit/gentree.cpp

```diff
@@ -24711,6 +24739,24 @@ bool GenTree::IsNeverNegative(Compiler* comp) const
     {
         return AsIntConCommon()->IntegralValue() >= 0;
     }
+
+    if (IsLocal())
```

Comment:
Do we really need given the integral range logic subsumes it? It doesn't look that common a case to warrant special-casing.

Additionally, this should only be valid for `LCL_VAR`s, not `LCL_FLD`s.

Reply:
> Do we really need given the integral range logic subsumes it

The other path is more expensive, less inlineable, and doing unnecessary extra work. This just shortcuts the check for these cases and simplifies the overall handling.

> Additionally, this should only be valid for LCL_VARs, not LCL_FLDs.

This is simply checking the relevant bitfield on the LclVarDsc. If the support is extended in the future then this will continue doing the right thing. If it isn't, then there is no issue because nothing will be setting or propagating the bit to `LCL_FLD` anyways.

Reply:
It is simply incorrect to ask this question for `LCL_FLD`s.

Consider:
```
LCL_FLD short [+0] V00 int
```
Even if `V00` always has the upper bit set to zero, it does not mean the 16th bit is always zero.

Reply:
Fixed